public/app/features/alerting/unified/api/alertmanager.ts

```diff
@@ -171,28 +171,52 @@ export async function testReceivers(
     receivers,
     alert,
   };
-  const result = await lastValueFrom(
-    getBackendSrv().fetch<TestReceiversResult>({
-      method: 'POST',
-      data,
-      url: `/api/alertmanager/${getDatasourceAPIId(alertManagerSourceName)}/config/api/v1/receivers/test`,
-      showErrorAlert: false,
-      showSuccessAlert: false,
-    })
-  );
+  try {
+    const result = await lastValueFrom(
+      getBackendSrv().fetch<TestReceiversResult>({
+        method: 'POST',
+        data,
+        url: `/api/alertmanager/${getDatasourceAPIId(alertManagerSourceName)}/config/api/v1/receivers/test`,
+        showErrorAlert: false,
+        showSuccessAlert: false,
+      })
+    );
 
-  // api returns 207 if one or more receivers has failed test. Collect errors in this case
-  if (result.status === 207) {
-    throw new Error(
-      result.data.receivers
-        .flatMap((receiver) =>
-          receiver.grafana_managed_receiver_configs
-            .filter((receiver) => receiver.status === 'failed')
-            .map((receiver) => receiver.error ?? 'Unknown error.')
-        )
-        .join('; ')
+    // api returns 207 if one or more receivers has failed test. Collect errors in this case
+    if (result.status === 207) {
```

Comment:
I think we can simplify a large part of this by checking for any errors in the response regardless of what the status code was.

Something like

```javascript
function receiversResponseContainsErrors(result: TestReceiversResult) {
  return result.receivers.some((receiver) =>
    receiver.grafana_managed_receiver_configs.some((config) => Boolean(config.error))
  );
}
```

should work for both `result.data` and `error.data`

Reply:
I adjusted the code. The only concern I have is about the shape of the `error.data` but the `isTestReceiverResult` type guard should be enough to handle that properly